lib/chef/resource/locale.rb

```diff
@@ -25,67 +25,115 @@ class Locale < Chef::Resource
       description "Use the locale resource to set the system's locale."
       introduced "14.5"
 
+      LC_VARIABLES = %w{LC_ADDRESS LC_COLLATE LC_CTYPE LC_IDENTIFICATION LC_MEASUREMENT LC_MESSAGES LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER LC_TELEPHONE LC_TIME}.freeze
+      LOCALE_CONF = "/etc/locale.conf".freeze
+      LOCALE_REGEX = /\A\S+/.freeze
+
       property :lang, String,
-               default: "en_US.utf8",
-               description: "Sets the default system language."
+               description: "Sets the default system language.",
+               regex: [LOCALE_REGEX],
+               validation_message: "The provided lang not valid. It should be a non-empty string without any leading whitespaces."
+
+      property :lc_env, Hash,
+               description: "A Hash of LC_* env variables in the form of ({'LC_ENV_VARIABLE' => 'VALUE'}).",
+               default: lazy { {} },
+               coerce: proc { |h|
+                         if h.respond_to?(:keys)
+                           invalid_keys = h.keys - LC_VARIABLES
+                           unless invalid_keys.empty?
+                             error_msg = "Key of option lc_env must be equal to one of: \"#{LC_VARIABLES.join('", "')}\"!  You passed \"#{invalid_keys.join(', ')}\"."
+                             raise Chef::Exceptions::ValidationFailed, error_msg
+                           end
+                         end
+                         unless h.values.all? { |x| x =~ LOCALE_REGEX }
+                           error_msg = "Values of option lc_env should be non-empty string without any leading whitespaces."
+                           raise Chef::Exceptions::ValidationFailed, error_msg
+                         end
+                         h
+                       }
 
-      property :lc_all, String,
-               default: "en_US.utf8",
-               description: "Sets the fallback system language."
+      # @deprecated Use {#lc_env} instead of this property.
+      #   {#lc_env} uses Hash with specific LC var as key.
+      # @raise [Chef::Deprecated]
+      #
+      def lc_all(arg = nil)
```

Comment:
We have a built in method for deprecating properties that should probably get used here.

Reply:
We used custom method here, because built in deprecation on property always shows [a specific URL](deprecations_locale_lc_all.html)

```
Deprecated features used!
  true at 1 location:
    - /recipes/default.rb:12:in `block in from_file'
   See https://docs.chef.io/deprecations_property.html for further details.
```

But we want to display a _custom URL here with different message_:
```
Deprecated features used!
  Changing LC_ALL can break Chef's parsing of command output in unexpected ways.
 Use one of the more specific LC_ properties as needed. at 1 location:
    - /recipes/default.rb:12:in `block in from_file'
   See https://docs.chef.io/deprecations_locale_lc_all.html for further details.
```

It seems like there is no way as of now by which we can pass custom URL and messages in deprecation on property itself.
 @tas50 Will it be fine if we could implement the same and achieve this.